aws/route_table.go

```diff
@@ -97,29 +97,44 @@ func (t *RouteTable) ListPossibleVips() *MaybeVips {
 	return &MaybeVips{t, ids, vips, names}
 }
 
-func (t *RouteTable) GetSrcInstanceByVip(vip string) (*Ec2Meta, error) {
+func (t *RouteTable) GetSrcByVip(vip string) (*Ec2Meta, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), timeOut)
 	defer cancel()
 
-	id, err := t.getSrcInstanceIdByVip(vip)
+	id, err := t.getSrcByVip(vip)
 	if err != nil {
 		return nil, err
 	}
-	name, err := t.cli.getInstanceNameById(ctx, id)
-	if err != nil {
-		return nil, err
+
+	var name string
+	switch {
+	case strings.HasPrefix(id, "i-"):
+		name, err = t.cli.getInstanceNameById(ctx, id)
+		if err != nil {
+			return nil, err
+		}
+	case strings.HasPrefix(id, "eni-"):
+		name, err = t.cli.getENINameById(ctx, id)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Other cases include Internet gateways and VPN gateways.  So, I try to handle these case by https://github.com/taku-k/swiro/pull/1/commits/23b8b628f9e22d75e48bb77520ad9d82e2fc8f36.